tests/test-network-transport/src/test/kotlin/testnetworktransport/MapTestNetworkTransportDispatcherTest.kt

```diff
@@ -0,0 +1,125 @@
+            path = listOf("hero", "name"),
+            extensions = null,
+            nonStandardFields = null))
+        ).build()
+
+    val query2 = GetHeroQuery("002")
+    val query2TestData = GetHeroQuery.Data(
+        GetHeroQuery.Hero(
+            __typename = "Droid",
+            name = "R2D2",
+            id = "r2d2",
+            onDroid = GetHeroQuery.OnDroid("mechadroid"),
+            onHuman = null
+        )
+    )
+    val testResponse2 = ApolloResponse.Builder(query2, uuid4(), query2TestData).build()
+
+    val query3 = GetHeroNameOnlyQuery()
+    val testResponse3 = ApolloResponse.Builder(query3, uuid4(), GetHeroNameOnlyQuery.Data(GetHeroNameOnlyQuery.Hero(name = "Darth Vader")))
+        .build()
+
+    val testNetworkTransport = apolloClient.testNetworkTransport
+    testNetworkTransport.register(query1, testResponse1)
+    testNetworkTransport.register(query2, testResponse2)
+    testNetworkTransport.register(query3, testResponse3)
+
+    val actual1: ApolloResponse<GetHeroQuery.Data> = apolloClient.query(query1).execute()
+    assertTrue(actual1.hasErrors())
+    assertEquals(testResponse1.errors!!.first().message, actual1.errors!!.first().message)
+
+    val actual2: ApolloResponse<GetHeroQuery.Data> = apolloClient.query(query2).execute()
+    assertFalse(actual2.hasErrors())
+    assertEquals(query2TestData, actual2.data)
+
+    val actual3: ApolloResponse<GetHeroNameOnlyQuery.Data> = apolloClient.query(query3).execute()
+    assertFalse(actual3.hasErrors())
+    assertEquals("Darth Vader", actual3.data!!.hero.name)
+  }
+
+  @Test
+  fun registerError() = runTest(before = { setUp() }, after = { tearDown() }) {
+    val query = GetHeroQuery("001")
+    apolloClient.testNetworkTransport.register(query, errors = listOf(Error(
+        message = "There was an error",
+        locations = listOf(Error.Location(line = 1, column = 2)),
+        path = listOf("hero", "name"),
+        extensions = mapOf("myExtension" to true),
+        nonStandardFields = null))
+    )
+
+    val actual: ApolloResponse<GetHeroQuery.Data> = apolloClient.query(query).execute()
+    assertTrue(actual.hasErrors())
+    assertEquals(actual.errors!!.first().message, "There was an error")
+    assertEquals(actual.errors!!.first().extensions!!["myExtension"], true)
+  }
+
+  @Test
+  fun registerDataManual() = runTest(before = { setUp() }, after = { tearDown() }) {
+    val query = GetHeroQuery("001")
+    val testData = GetHeroQuery.Data(
+        GetHeroQuery.Hero(
+            __typename = "Droid",
+            name = "R2D2",
+            id = "r2d2",
+            onDroid = GetHeroQuery.OnDroid("mechadroid"),
+            onHuman = null
+        )
+    )
+    apolloClient.testNetworkTransport.register(query, testData)
+
+    val actual = apolloClient.query(query).execute().data
+    assertEquals(testData, actual)
+  }
+
+  @Test
+  fun registerDataTestBuilder() = runTest(before = { setUp() }, after = { tearDown() }) {
```

Comment:
I don't think we need both `Manual` and `TestBuilder` versions of this test. How the `data` is constructed is tested in other tests already.